### PR TITLE
[Balance] Egg and Starter Adjustments

### PR DESCRIPTION
**Pull request by damocleas**
## What are the changes?
Adjusted the chances of HA in Voucher-claimed Eggs.
Adjusted the chances of HA and Shiny in Same-Species Eggs you buy with candy.
Adjusted costs of Same-Species Eggs.
Adjusted Passive and Cost Reductions for the far ends of point costs.
Matched Hisuian Pokemon to be the same rarity in eggs as other regional pokemon.

## Why am I doing these changes?
Extremely underwhelming as is now, feels like a chore / grinding simulator for all of these mechanics, and adding a real way to unlock Hidden Abilities is big, even if still a bit of a grind.

## What did change?
Voucher Egg Changes:
- Hidden Ability Rate 1/256 -> 1/192

Same-Species Egg Chances:
- Shiny Rate 1/32 -> 1/24
- Hidden Ability Rate 1/16 -> 1/4

Same-Species Egg Costs Reduced for all values:
- 1 / 35 -> 30
- 2 / 35 -> 30
- 3 / 35 -> 25
- 4 / 30 -> 20
- 5 / 25 -> 18
- 6 / 20 -> 15
- 7 / 15 -> 12
- 8 / 10 -> 8
- 9 / 10 -> 8
- 10 / 10 -> 8

Passive Cost Changes:
- 1 / 50 -> 40
- 2 / 45 -> 40
- 3 / 40 -> 35

Cost Reduction Changes:
- 1 / 30, 75 -> 25, 60
- 9 / 3, 10 -> 5, 15
- 10 / 3, 10 -> 5, 15

Regional Pokemon from Eggs Changes:
- Hisuian Pokemon 7/8 reduction -> 1/2 reduction to match other regionals

## How to test the changes?
Merge with local branch.

## Checklist
- [X] **I'm using `beta` as my base branch**
- [X] There is no overlap with another PR?
- [X] The PR is self-contained and cannot be split into smaller PRs?
- [X] Have I provided a clear explanation of the changes?
- [X] Have I considered writing automated tests for the issue?

### CODE REVIEW

File: src/data/egg.ts [hunk 1]
```diff
@@ -408,7 +407,7 @@ export class Egg {
       let weight = Math.floor((((maxStarterValue - speciesStarters[speciesId]) / ((maxStarterValue - minStarterValue) + 1)) * 1.5 + 1) * 100);
       const species = getPokemonSpecies(speciesId);
       if (species.isRegional()) {
-        weight = Math.floor(weight / (species.isRareRegional() ? 8 : 2));
+        weight = Math.floor(weight / (species.isRareRegional() ? 2 : 1));
```
Comment (Tempo-anon) [inline src/data/egg.ts:410]: This just sets only Hisui ones to 0.5x weight while other regionals to 1x

Comment (damocleas) [reply src/data/egg.ts:410]: What do I do to make them all .5x or even just the same as every other mon then?

Comment (Tempo-anon) [reply src/data/egg.ts:410]:
```ts
if (species.isRegional()) {
  weight = Math.floor(weight / 2);
}
```

**Comment by DayKev** [comment]
Made a PR with the HA changes discussed on Discord. https://github.com/damocleas/pokerogue/pull/1

**Pull request merged**